helm-charts/splunk-otel-collector/templates/_helpers.tpl

```diff
@@ -457,3 +457,12 @@ Build the securityContext for Linux and Windows
 {{- end }}
 {{- toYaml .securityContext }}
 {{- end -}}
+
+
+{{- define "splunk-otel-collector.logsEngine" -}}
```

Comment:
Nit: Just to be more direct I would suggest this update to protect from values like " " or "Fluentd".

{{- define "splunk-otel-collector.logsEngine" -}}
{{- if eq .Values.logsEngine "fluentd" -}}
fluentd
{{- else if eq .Values.logsEngine "otel" -}}
otel
{{- else -}}
{{- fail "Invalid logsEngine value provided. Use either 'fluentd' or 'otel'." -}}
{{- end -}}
{{- end -}}

Reply:
This is already protected by values.schema.json